src/WidgetBase.ts

```diff
@@ -239,6 +255,24 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 		}));
 	}
 
+	protected meta<T extends WidgetMeta>(MetaType: WidgetMetaConstructor<T>): T {
+		let cached = this._metaMap.get(MetaType);
+		if (!cached) {
+			cached = new MetaType({
+				nodes: this._nodeMap,
+				invalidate: this.invalidate.bind(this)
+			});
+
+			this._metaMap.set(MetaType, cached);
+		}
+
+		if (cached.requiresRender && !this._cachedVNode) {
```

Comment:
OK, removed `requiresRender` completely and we're relying on the passed in `invalidate` property. If a widget requires a node it can call `invalidate` itself.